AppCenterAnalytics/AppCenterAnalytics/Internals/Model/Properties/MSTypedProperty.m

```diff
@@ -0,0 +1,50 @@
+#import "MSTypedProperty.h"
+#import "MSAnalyticsInternal.h"
+#import "MSConstants+Internal.h"
+#import "MSLogger.h"
+
+static NSString *const kMSTypedPropertyType = @"type";
+
+static NSString *const kMSTypedPropertyName = @"name";
+
+@implementation MSTypedProperty
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+    self = [super init];
+    if (self) {
+        _type = [coder decodeObjectForKey:kMSTypedPropertyType];
+        _name = [coder decodeObjectForKey:kMSTypedPropertyName];
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+    [coder encodeObject:self.type forKey:kMSTypedPropertyType];
+    [coder encodeObject:self.name forKey:kMSTypedPropertyName];
+}
+
+/**
+ * Serialize this object to a dictionary.
+ *
+ * @return A dictionary representing this object.
+ */
+- (NSMutableDictionary *)serializeToDictionary {
+    NSMutableDictionary * dict = [NSMutableDictionary new];
+    dict[kMSTypedPropertyType] = self.type;
+    dict[kMSTypedPropertyName] = self.name;
+    return dict;
+}
+
+- (instancetype)createValidCopyForAppCenter {
+    if ([self.name length] > 125) {
```

Comment:
The property key validation can be shared between string only `properties` and `typedProperties` once moved to `Analytics+Validation`